core/api/src/app/payments/send-intraledger.ts

```diff
@@ -236,151 +223,208 @@ const executePaymentViaIntraledger = async <
-    })
-    if (recipientDisplayPriceRatio instanceof Error) return recipientDisplayPriceRatio
-    const { displayAmount: recipientDisplayAmount, displayFee: recipientDisplayFee } =
-      DisplayAmountsConverter(recipientDisplayPriceRatio).convert(paymentFlow)
+        paymentFlow,
+        senderDisplayCurrency: senderAccount.displayCurrency,
+        senderUsername: senderAccount.username,
+        recipientDisplayCurrency: recipientAccount.displayCurrency,
+        recipientUsername: recipientAccount.username,
 
-    if (signal.aborted) {
-      return new ResourceExpiredLockServiceError(signal.error?.message)
-    }
+        memo,
 
-    let metadata:
-      | AddWalletIdIntraledgerSendLedgerMetadata
-      | AddWalletIdTradeIntraAccountLedgerMetadata
-    let additionalDebitMetadata: {
-      [key: string]: Username | DisplayCurrencyBaseAmount | DisplayCurrency | undefined
-    } = {}
-    let additionalCreditMetadata: {
-      [key: string]: DisplayCurrencyBaseAmount | DisplayCurrency | undefined
-    }
-    let additionalInternalMetadata: {
-      [key: string]: DisplayCurrencyBaseAmount | DisplayCurrency | undefined
-    } = {}
-    if (senderWallet.accountId === recipientWallet.accountId) {
-      ;({
-        metadata,
-        debitAccountAdditionalMetadata: additionalDebitMetadata,
-        creditAccountAdditionalMetadata: additionalCreditMetadata,
-        internalAccountsAdditionalMetadata: additionalInternalMetadata,
-      } = LedgerFacade.WalletIdTradeIntraAccountLedgerMetadata({
-        paymentAmounts: paymentFlow,
-
-        senderAmountDisplayCurrency: toDisplayBaseAmount(senderDisplayAmount),
-        senderFeeDisplayCurrency: toDisplayBaseAmount(senderDisplayFee),
-        senderDisplayCurrency: senderDisplayCurrency,
-
-        memoOfPayer: memo || undefined,
-      }))
-    } else {
-      ;({
-        metadata,
-        debitAccountAdditionalMetadata: additionalDebitMetadata,
-        creditAccountAdditionalMetadata: additionalCreditMetadata,
-        internalAccountsAdditionalMetadata: additionalInternalMetadata,
-      } = LedgerFacade.WalletIdIntraledgerLedgerMetadata({
-        paymentAmounts: paymentFlow,
-
-        senderAmountDisplayCurrency: toDisplayBaseAmount(senderDisplayAmount),
-        senderFeeDisplayCurrency: toDisplayBaseAmount(senderDisplayFee),
-        senderDisplayCurrency: senderDisplayCurrency,
-
-        recipientAmountDisplayCurrency: toDisplayBaseAmount(recipientDisplayAmount),
-        recipientFeeDisplayCurrency: toDisplayBaseAmount(recipientDisplayFee),
-        recipientDisplayCurrency: recipientAccount.displayCurrency,
+        senderAsNotificationRecipient,
+        recipientAsNotificationRecipient,
+      }),
+  )
+  if (paymentSendResult instanceof Error) return paymentSendResult
 
-        memoOfPayer: memo || undefined,
-        senderUsername: senderAccount.username,
-        recipientUsername,
-      }))
-    }
+  if (senderAccount.id !== recipientAccount.id) {
```

Comment:
not a blocker but... moving this inside `executePaymentViaIntraledger` is a bit confusing, actually this is not part of payment execution (the same applies to notifications but it is more critical because of the callbacks)

cc @nicolasburtey @bodymindarts

Reply:
I did this more to be consistent with the `executePaymentViaIntraledger` function in `send-lightning` but can move both to be outside the function if that makes more sense

Reply:
Kk fixed here in both files https://github.com/GaloyMoney/galoy/pull/4181/commits/a0e1fd49af9d844cd8e1b97e3b826e5337d4e6ad

Reply:
Also I think you're right about this being a distinct interface being clearer if it weren't in helpers. Moved it to `accounts` module here https://github.com/GaloyMoney/galoy/pull/4181/commits/ba94cbbb40be988f046386b1e2de4c24063c2811